conan/tools/cmake/toolchain/blocks.py

```diff
@@ -113,6 +113,54 @@ def context(self):
         return {"vs_runtimes": config_dict}
 
 
+class VSDebuggerEnvironment(Block):
+    template = textwrap.dedent("""
+        {% if vs_debugger_path %}
+        # Definition of CMAKE_VS_DEBUGGER_ENVIRONMENT
+        set(CMAKE_VS_DEBUGGER_ENVIRONMENT "{{ vs_debugger_path }}")
+        {% endif %}
+        """)
+
+    def context(self):
+        os_ = self._conanfile.settings.get_safe("os")
+        build_type = self._conanfile.settings.get_safe("build_type")
+
+        if (os_ and "Windows" not in os_) or not build_type:
+            return None
+
+        if "Visual" not in self._toolchain.generator:
+            return None
+
+        config_dict = {}
+        if os.path.exists(CONAN_TOOLCHAIN_FILENAME):
+            existing_include = load(CONAN_TOOLCHAIN_FILENAME)
+            pattern = r"set\(CMAKE_VS_DEBUGGER_ENVIRONMENT \"PATH=([^)]*);%PATH%\"\)"
+            vs_debugger_environment = re.search(pattern, existing_include)
+            if vs_debugger_environment:
+                capture = vs_debugger_environment.group(1)
+                matches = re.findall(r"\$<\$<CONFIG:([A-Za-z]*)>:([^>]*)>", capture)
+                config_dict = dict(matches)
+
+        host_deps = self._conanfile.dependencies.host.values()
+        test_deps = self._conanfile.dependencies.test.values()
+        bin_dirs = [p for dep in host_deps for p in dep.cpp_info.aggregated_components().bindirs]
+        test_bindirs = [p for dep in test_deps for p in dep.cpp_info.aggregated_components().bindirs]
+        bin_dirs.extend(test_bindirs)
+        bin_dirs = [p.replace("\\", "/") for p in bin_dirs]
+
+        bin_dirs = ";".join(bin_dirs)
+        config_dict[build_type] = bin_dirs
+        if all(value is '' for value in config_dict.values()):
+            return None
```

Comment:
Is this possible? Could we have this `bindirs = ["", "bin"]`? If not, I think it'd be better:

```suggestion
        if not bin_dirs:
            return None

        bin_dirs = ";".join(bin_dirs)
        config_dict[build_type] = bin_dirs
```